### DISCUSSION TITLE
How to align the formula number to the right in docx output

### DISCUSSION THREAD

**Discussion by Dagwbl** [quarto-dev/quarto-cli#2618] [Q&A]
I have to submit in docx format, then I encountered a problem, That is aligning the formula number to the right.
the quarto default style like this
![image](https://user-images.githubusercontent.com/39078825/192413524-dba6b0c5-acf0-4b27-a415-ca134b3963aa.png)

But I want to the style like this
![image](https://user-images.githubusercontent.com/39078825/192413669-27613765-159c-4e5f-bfa6-f487bde270e4.png)

 In native pandoc, I know we can use the `--crossref` and add template to YAML, but in quarto, I don't know what to do?

**Comment by cderv** [comment]
>  I know we can use the --crossref and add template to YAML, but in quarto, I don't know what to do?

You can still use template with Quarto (https://quarto.org/docs/output-formats/ms-word-templates.html) and crossref is built in quarto (https://quarto.org/docs/authoring/cross-references.html#equations) 

Did you try already the template ? Does the Pandoc crossref filter you used handled that differently that Quarto ? 

If you could share a full reproducible example that would help

**Reply by Dagwbl** [reply]
In order to achieve the right side of the formula numbers, we can use `pandoc-crossref` in native pandoc.
As shown in the following md
```markdown
---
autoEqnLabels: true 
tableEqns: true 
eqnBlockTemplate: |
   `<w:pPr><w:tabs><w:tab w:val="center" w:leader="none" w:pos="4819" /><w:tab w:val="right" w:leader="none" w:pos="9638" /></w:tabs></w:pPr><w:r><w:tab /></w:r>`{=openxml} $$t$$ `<w:r><w:tab /></w:r>`{=openxml} $$i$$

eqnBlockInlineMath: true
equationNumberTeX: \\tag
eqnIndexTemplate: ($$i$$) 
eqnPrefixTemplate: Eq($$i$$) 
---
# Test formula numbers alignment
[@eq:eq2]
$$
x+y=z
$${#eq:eq1}
formula 1

$$
x+y=z
$${#eq:eq2}
formula 2

$$
x+y=z
$${#eq:eq3}
formula 3

$$
x+y=z
$${#eq:eq4}
```

The conversion command is as follows：
```bash
pandoc -F pandoc-crossref test.md -o test.docx
```
Results:
![image](https://user-images.githubusercontent.com/39078825/192524079-0f33260e-d0f2-4692-bdd1-0d7a4c63f0b2.png)
We can compare with the default style of quarto.
 
![image](https://user-images.githubusercontent.com/39078825/192524466-5330045c-6f24-4a9a-928e-2b523fb914fe.png)

We can use native `pandoc` with `pandoc-crossref` to implement that the formula number align to the right of the page, but this requires a series of `pandoc-crossref` parameters to be set. 

I don't know how to set the `pandoc-crossref` parameters at `quarto`.

**Reply by Dagwbl** [reply]
> 
I tried the template, but I found that the formula did not apply a style

**Reply by cderv** [reply]
Oh ok, it is not the same syntax as `pandoc-crossref` and we don't have the feature of setting all this 
```
autoEqnLabels: true 
tableEqns: true 
eqnBlockTemplate: |
   `<w:pPr><w:tabs><w:tab w:val="center" w:leader="none" w:pos="4819" /><w:tab w:val="right" w:leader="none" w:pos="9638" /></w:tabs></w:pPr><w:r><w:tab /></w:r>`{=openxml} $$t$$ `<w:r><w:tab /></w:r>`{=openxml} $$i$$

eqnBlockInlineMath: true
equationNumberTeX: \\tag
eqnIndexTemplate: ($$i$$) 
eqnPrefixTemplate: Eq($$i$$) 
```

I believe you can set 
```
crossref: false
```
in the YAML metadata  to deactivate cross ref mechanism from Quarto and use `pandoc-crossref` instead by adding to `filters`. 


@jjallaire is this some feature we plan to make closer to `pandoc-crossref` ? Do we need to track this in a specific feature request ?

**Reply by Dagwbl** [reply]
Thank you for the solution. I still expect quarto to implement the small detail. After all, in html or pdf, the formula numbers are aligned on the right, and I think they should be consistent.
